source/s3_auto_ranged_get.c

```diff
@@ -182,6 +182,39 @@ static void s_s3_auto_ranged_get_next_request(
         }
     }
 
+    /* One sepcial error case is an empty file download */
+    struct aws_s3_meta_request_result finish_result = meta_request->synced_data.finish_result;
+    if (finish_result.error_response_body && finish_result.error_response_headers) {
+        struct aws_byte_cursor content_type;
+        AWS_ZERO_STRUCT(content_type);
+        if (!aws_http_headers_get(
+                finish_result.error_response_headers, aws_byte_cursor_from_c_str("content-type"), &content_type)) {
+            /* Content type found */
+            if (aws_byte_cursor_eq_c_str_ignore_case(&content_type, "application/xml")) {
+                /* XML response */
+                struct aws_byte_cursor body_cursor = aws_byte_cursor_from_buf(finish_result.error_response_body);
+                struct aws_byte_cursor tag = aws_byte_cursor_from_c_str("ActualObjectSize");
+                struct aws_string *size = get_top_level_xml_tag_value(meta_request->allocator, &tag, &body_cursor);
+                if (aws_string_eq_c_str(size, "0")) {
```

Comment:
you could store the comparison result and then destroy the string before checking it.  Then you wouldn't have to have multiple destroys in different branches.